ground/src/main/java/com/google/android/ground/ui/common/LocationOfInterestHelper.kt

```diff
@@ -17,46 +17,58 @@ package com.google.android.ground.ui.common
 
 import android.content.res.Resources
 import com.google.android.ground.R
-import com.google.android.ground.model.AuditInfo
-import com.google.android.ground.model.User
-import com.google.android.ground.model.geometry.LineString
-import com.google.android.ground.model.geometry.LinearRing
+import com.google.android.ground.model.geometry.Geometry
 import com.google.android.ground.model.geometry.MultiPolygon
 import com.google.android.ground.model.geometry.Point
 import com.google.android.ground.model.geometry.Polygon
 import com.google.android.ground.model.locationofinterest.LocationOfInterest
-import java.util.Optional
 import javax.inject.Inject
 
-/** Common logic for formatting attributes of [LocationOfInterest] for display to the user. */
+/** Helper class for creating user-visible text. */
 class LocationOfInterestHelper @Inject internal constructor(private val resources: Resources) {
-  fun getCreatedBy(locationOfInterest: Optional<LocationOfInterest>): String =
-    getUserName(locationOfInterest).map { resources.getString(R.string.added_by, it) }.orElse("")
 
-  // TODO(#793): Allow user-defined LOI names for other LOI types.
-  fun getLabel(locationOfInterest: Optional<LocationOfInterest>): String =
-    locationOfInterest.map(::getLabel).orElse("")
+  fun getDisplayLoiName(loi: LocationOfInterest): String {
+    val loiId = loi.customId.ifEmpty { loi.getProperty("id") }
+    val loiName = loi.getProperty("name")
 
-  fun getLabel(loi: LocationOfInterest): String {
-    // TODO(#2046): Reuse logic from card util to display LOI label
-    val caption = loi.customId?.trim { it <= ' ' } ?: ""
-    return caption.ifEmpty { getLocationOfInterestType(loi) }
+    return if (loiName.isNotEmpty() && loiId.isNotEmpty()) {
+      "$loiName ($loiId)"
+    } else if (loiName.isNotEmpty()) {
+      loiName
+    } else if (loiId.isNotEmpty()) {
+      "${loi.geometry.toType()} ($loiId)"
+    } else {
+      loi.geometry.toDefaultName()
+    }
   }
 
-  private fun getLocationOfInterestType(locationOfInterest: LocationOfInterest): String =
-    when (locationOfInterest.geometry) {
-      is Polygon -> "Polygon"
-      is Point -> "Point"
-      is LineString -> "LineString"
-      is LinearRing -> "LinearRing"
-      is MultiPolygon -> "MultiPolygon"
+  fun getJobName(loi: LocationOfInterest): String? = loi.job.name
+
+  fun getSubmissionsText(count: Int): String =
+    when (count) {
+      0 -> "No submissions"
+      1 -> "$count submission"
+      else -> "$count submissions"
```

Comment:
how about bringing these hardcoded values from strings.xml, this way it will be more easy to modify in future.

Reply:
Thanks for the suggestion! Done